src/python/TaskWorker/Actions/PreJob.py

```diff
@@ -298,6 +301,16 @@ def alter_submit(self, crab_retry):
         ## Add the site black- and whitelists and the DESIRED_SITES to the
         ## Job.<job_id>.submit content.
         new_submit_text = self.redo_sites(new_submit_text, crab_retry, use_resubmit_info)
+
+        ## Add group information:
+        username = self.task_ad.get('CRAB_UserHN')
+        if 'CMSGroups' in self.task_ad:
+            new_submit_text += '+CMSGroups = %s\n' % classad.quote(groups)
```

Comment:
I think groups is undefined. Did you miss a `groups = self.task_ad.get('CMSGroups')`?

I will try to run the code after dinner before merging it.